onload/cplane/main.go

```diff
@@ -0,0 +1,133 @@
+// SPDX-License-Identifier: MIT
+// SPDX-FileCopyrightText: (c) Copyright 2023 Advanced Micro Devices, Inc.
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/golang/glog"
+
+	corev1 "k8s.io/api/core/v1"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+)
+
+const (
+	cplaneServerPathFilePath       = "/sys/module/onload/parameters/cplane_server_path"
+	cplaneServerParamsFilePath     = "/sys/module/onload/parameters/cplane_server_params"
+	critclFilePath                 = "/usr/bin/crictl"
+	crioPrefix                     = "cri-o://"
+	cplaneServerParamsFormatString = "exec %s /opt/onload/sbin/onload_cp_server -K"
+
+	timeoutTime = 5 * time.Second
+)
+
+var (
+	podName      string
+	podNamespace string
+
+	containerID string
+)
+
+func main() {
+
+	podName = os.Getenv("POD_NAME")
+	podNamespace = os.Getenv("POD_NAMESPACE")
+
+	cfg, err := config.GetConfig()
```

Comment:
Official [example](https://github.com/kubernetes/client-go/blob/master/examples/in-cluster-client-configuration/main.go#L41) uses `rest.InClusterConfig()` from `k8s.io/client-go/rest` library. Is there a reason the controller library is required here instead?

Reply:
My previous understanding was that the kubernetes controller library was necessary for getting the controller ID, but obviously that is not the case.
I have just tested it using the `client-go` library and it works, so in a second I'll update the PR to use as I think it would be more appropriate.